### PR TITLE
updates to exercise READMEs

### PR DESCRIPTION
**Pull request by rpottsoh**
The main purpose of this PR is to point out that there are 18 exercises who's READMEs are out of date.  

Please pay special attention to Bob as he is using a local (.meta) copy of description.md that is not current with exercism/problem-specifications.  The .meta version of description.md supersedes the version in problem-specifications.

`configlet generate` was used to create these updates.

my curiosity was triggered by exercism/exercism#4292

I am just attempting to be helpful and don't want to interfere.

### CODE REVIEW

**Comment by tejasbubane** [comment]
Regarding `bob`, I think we can sync up the upstream specification and remove `.meta` - but that can be done separately. Waiting to see if others have any issues.

File: exercises/forth/README.md [hunk 1]
```diff
@@ -52,3 +52,7 @@ In the test suites all tests but the first have been skipped.
 
 Once you get a test passing, you can enable the next one by
 changing `xtest` to `test`.
+
+
+## Submitting Incomplete Solutions
+It's possible to submit an incomplete solution so you can see how others have completed the exercise.
```
Comment (tejasbubane) [inline exercises/forth/README.md]: Any reason for this to be in added specifically for this one exercise?

File: exercises/hamming/README.md [hunk 2]
```diff
@@ -31,9 +31,10 @@ The Hamming distance between these two DNA strands is 7.
 
 # Implementation notes
 
-The Hamming distance is only defined for sequences of equal length. This means
-that based on the definition, each language could deal with getting sequences
-of equal length differently.
+The Hamming distance is only defined for sequences of equal length, so
+an attempt to calculate it between sequences of different lengths should
+not work. The general handling of this situation (e.g., raising an
+exception vs returning a special value) may differ between languages.
```
Comment (tejasbubane) [inline exercises/hamming/README.md]: Can you update this to the implementation in this track? We `throw and error` in this case: https://github.com/exercism/javascript/blob/master/exercises/hamming/hamming.spec.js#L34

File: exercises/nucleotide-count/README.md [hunk 3]
```diff
@@ -1,34 +1,23 @@
 # Nucleotide Count
 
-Given a single stranded DNA string, compute how many times each nucleotide occurs in the string. Also if the
-given input strand contains invalid nucleotides, throw the error "Invalid nucleotide in strand"
+Given a single stranded DNA string, compute how many times each nucleotide occurs in the string.
```
Comment (tejasbubane) [inline exercises/nucleotide-count/README.md]: we throw an error for `invalid nucleotide`. This testcase needs to be removed: https://github.com/exercism/javascript/blob/master/exercises/nucleotide-count/nucleotide-count.spec.js#L21

**Comment by tejasbubane** [comment]
Left some comments. We need to sync up soon with canonical test cases one exercise at a time in separate PRs. But these README changes should be fine to go in.

File: exercises/forth/README.md (hunk 1 above)
Comment (rpottsoh) [reply exercises/forth/README.md]: This is coming from `config/exercise_readme.go.tmpl`.  You are not seeing it already in other READMEs?

File: exercises/hamming/README.md (hunk 2 above)
Comment (rpottsoh) [reply exercises/hamming/README.md]: It appears like the testsuite may be behind the testdata.  The testdata has more than one case where an exception my be raised.  
see:
https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json#L135
and
https://github.com/exercism/problem-specifications/blob/master/exercises/hamming/canonical-data.json#L144

File: exercises/nucleotide-count/README.md (hunk 3 above)
Comment (rpottsoh) [reply exercises/nucleotide-count/README.md]: Are you asking to have nucleotide-count/README.md changes removed from this PR?

**Comment by rpottsoh** [comment]
@tejasbubane I am not clear if you would like some changes to be made to this PR.  You may not want the Bob changes since it is being swayed from the problem-specification version via. .meta data.  Let me know.  I can easily do nothing. :smile:

File: exercises/forth/README.md (hunk 1 above)
Comment (tejasbubane) [reply exercises/forth/README.md]: Ohh right, did not check other READMEs. The PR diff made me think it was added only for this exercise.

File: exercises/nucleotide-count/README.md (hunk 3 above)
Comment (tejasbubane) [reply exercises/nucleotide-count/README.md]: This was a note-to-self point to be done in another PR. Sorry for the confusion.

File: exercises/hamming/README.md (hunk 2 above)
Comment (tejasbubane) [reply exercises/hamming/README.md]: Ok let me quickly do the test-case updation in another PR and you go ahead and update the README according to canonical implementation in this PR. I just did not want the `may differ between languages` part to remain there.

**Comment by rpottsoh** [comment]
> Regarding bob, I think we can sync up the upstream specification and remove .meta - but that can be done separately. Waiting to see if others have any issues.

I could remove the update for Bob (the 2nd commit) from this PR.  Less churn on that README.

File: exercises/hamming/README.md (hunk 2 above)
Comment (tejasbubane) [reply exercises/hamming/README.md]: Updated: https://github.com/exercism/javascript/pull/458

**Comment by tejasbubane** [comment]
> I could remove the update for Bob (the 2nd commit) from this PR. Less churn on that README.

I think we should keep `bob` changes in separate PR?

**Comment by rpottsoh** [comment]
I will remove the Bob commit since it doesn't make the README current anyway.

**Comment by rpottsoh** [comment]
Bob's gone

**Comment by tejasbubane** [comment]
Thanks! @rpottsoh

**Pull request merged**